include/motor-control/core/linear_motion_system.hpp

```diff
@@ -24,11 +24,13 @@ concept MotorMechanicalConfig = requires {
     std::is_same_v<MC, BeltConfig> || std::is_same_v<MC, LeadScrewConfig>;
 };
 
+// template <MotorEncoderConfig ECConfig>
 template <MotorMechanicalConfig MEConfig>
 struct LinearMotionSystemConfig {
     MEConfig mech_config{};
     float steps_per_rev{};
     float microstep{};
+    float encoder_ppr{};  // pulses_per_revolution
```

Comment:
Small nit: if you specify the default value as 0.0 as in line 34 for gear ratio, then you don't have to write out encoder_ppr=0 for those systems that doesn't actually have an encoder, i.e. gripper z axis